src/firewall-applet.in

```diff
@@ -603,8 +605,15 @@ class TrayApplet(QtWidgets.QSystemTrayIcon):
     def quit(self):
         sys.exit(1)
 
-    def set_icon(self, mode):
-        self.setIcon(self.icons[mode])
+    def set_icon(self, mode=None):
+        if mode is not None:
+            self.setIcon(self.icons[mode])
+        elif self.mode != "normal":
+            self.setIcon(self.icons[self.mode])
+        elif self.default_zone == self.shields_up:
+            self.setIcon(self.icons["normal-shields_up"])
+        else:
+            self.setIcon(self.icons["normal-shields_down"])
```

Comment:
Does it makes sense to use the "normal" icon for "shields down".

Reply:
In case the `firewall-applet-normal-shields_down` icon does not exist in the current theme, it would fall back to the `firewall-applet-normal` icon.
See https://specifications.freedesktop.org/icon-naming-spec/icon-naming-spec-latest.html#guidelines (second bullet point) for details

Reply:
Ah, I follow you now. In that case this patch is functional as is. If you want I can merge this and you can do a separate PR for a new "shields up" icon.

Reply:
Please do - this allows to decouple code from artwork decisions